app/services/request_reference.rb

```diff
@@ -1,4 +1,18 @@
 class RequestReference
+  include ActiveModel::Model
+  attr_accessor :reference
+
+  validate :reference_complete_information, if: :new_reference_flow?
+
+  def send_request
+    return unless valid?
+
+    # Backwards compatibility between old and new references
+    # because I need to validate the @reference object
+    call(@reference)
```

Comment:
given that `reference` is an instance variable on this object, why pass it around?

Reply:
The call method is used in the old references (and it was used in the new as well), so I choose the path of not touching on the old call method. Just extend the behaviour I need for the new references (validation) and calling the method #call that is used in both old and new references.